benchmark/utils/general.hpp

```diff
@@ -392,6 +392,22 @@ gko::remove_complex<ValueType> compute_norm2(const vec<ValueType> *b)
 }
 
 
+template <typename ValueType>
+gko::remove_complex<ValueType> compute_direct_error(const gko::LinOp *solver,
+                                                    const vec<ValueType> *b,
+                                                    const vec<ValueType> *x)
+{
+    auto ref_exec = gko::ReferenceExecutor::create();
+    auto exec = solver->get_executor();
+    auto ref_solver = gko::clone(ref_exec, solver);
+    auto one = gko::initialize<vec<ValueType>>({1.0}, exec);
+    auto neg_one = gko::initialize<vec<ValueType>>({-1.0}, exec);
+    auto res = gko::clone(ref_exec, x);
+    ref_solver->apply(lend(one), lend(b), lend(neg_one), lend(res));
+    return compute_norm2(lend(res));
```

Comment:
It would be nice to just return the error vector here as a unique_ptr, and compute the norm in the `if(is_direct_solver)` block in the solver.cpp file. In some cases, it might be of interest to have access to the actual error vector.